google/gax/api_callable.py

```diff
@@ -141,49 +142,48 @@ def _bundleable(a_func, desc, bundler):
     bundled call.
 
     Args:
-        a_func (callable[[req], resp]): an API call that supports bundling.
+        a_func (callable[[req, options], resp]): an API call that supports
+          bundling.
+        settings (gax.CallSettings): the base call settings of the behavior.
         desc (gax.BundleDescriptor): describes the bundling that a_func
           supports.
-        bundler (gax.bundling.Executor): orchestrates bundling.
 
     Returns:
         callable: takes the API call's request and keyword args and returns a
           bundling.Event object.
 
     """
-    def inner(*args, **kwargs):
+    def inner(request, options=None, *args, **kwargs):
         """Schedules execution of a bundling task."""
-        request = args[0]
+        bundler = settings.merge(options).bundler
+        if not bundler:
+            return a_func(request, options, *args, **kwargs)
+
         the_id = bundling.compute_bundle_id(
             request, desc.request_discriminator_fields)
         return bundler.schedule(a_func, the_id, desc, request, kwargs)
 
     return inner
 
 
-def _page_streamable(a_func, page_descriptor, page_token=None,
-                     flatten_pages=True):
+def _page_streamable(a_func, settings, page_descriptor):
```

Comment:
The `page_descriptor` parameter is redundant with the `settings` parameter -- let's just use `settings`